File: 3. Data Science & Algorithms/Sort Algorithms/Insertion Sort/InsertionSort.py
def InsertionSort(array):
    first_unsorted_index= 1
    while first_unsorted_index < len(array):
        new_element= array[first_unsorted_index]
        new_index= first_unsorted_index

        while new_index > 0 and array[new_index-1] > new_element:
            array[new_index] = array[new_index-1]
            new_index -= 1

        array[new_index] = new_element

        first_unsorted_index += 1

    return array

print(InsertionSort([20, 35, -15, 7,  55, 1, -22]))

File: 3. Data Science & Algorithms/Sort Algorithms/Insertion Sort/InsertionSort.java
package com.company;

public class InsertionSort {

    public static void main(String[] args)  {

        int[] intArray = {20, 35, -15, 7,  55, 1, -22};

        for (int firstUnsortedIndex = 1; firstUnsortedIndex < intArray.length; firstUnsortedIndex++)    {

            int newElement = intArray[firstUnsortedIndex];
            int newIndex;

            for (newIndex = firstUnsortedIndex; newIndex > 0 && intArray[newIndex-1] > newElement; newIndex--)    {
                intArray[newIndex] = intArray[newIndex-1];
            }

            intArray[newIndex] = newElement;

        }

        for (int i = 0; i < intArray.length; i++)   {
            System.out.println(intArray[i]);
        }

    }

}
